### ISSUE TITLE
No visible cursor

### ISSUE THREAD

**Issue by jazzzz** [prismagraphql/graphql-playground#790]
#### This issue pertains to the following package(s):

- [ ] GraphQL Playground - Electron App
- [x] GraphQL Playground HTML
- [ ] GraphQL Playground
- [ ] GraphQL Playground Express Middleware
- [ ] GraphQL Playground Hapi Middleware
- [ ] GraphQL Playground Koa Middleware
- [ ] GraphQL Playground Lambda Middleware

#### What OS and OS version are you experiencing the issue(s) on?

Ubuntu 18.04 (under Firefox and Chrome).

#### What version of graphql-playground(-electron/-middleware) are you experiencing the issue(s) on?

graphql-playground-html 1.6.0 (as a dependency of apollo-server).

#### What is the expected behavior?

A cursor is displayed where I input the text.

#### What is the actual behavior?

No cursor is visible, I don't know where the text I type will be inserted.
In the HTML, the cursor is present and blinking, but with transparent background-color.

#### What steps may we take to reproduce the behavior?

Open GraphQL Playground.

_Please provide a gif or image of the issue for a quicker response/fix._

(it's a static image, but a video gives the same result)
![image](https://user-images.githubusercontent.com/290626/42786831-08d7d11e-8958-11e8-9ff5-f85c2c333300.png)

**Comment by gajus** [comment]
Inspecting the CSS, the element with the `CodeMirror-cursors` class does not have any attributes defining either width, height or color of the element. It appears that some CSS is getting swallowed somewhere. This should be marked as a bug @timsuchanek.

**Comment by vnenkpet** [comment]
+1 
Really inconvenient to work with this.

**Comment by jaywhy13** [comment]
Is there a known workaround for this as yet?

**Comment by vinicius73** [comment]
Any news abou this?

**Comment by ryanwiesjahn** [comment]
Here's a work-around that I'm using:

Using the browser extension [Stylus](https://github.com/openstyles/stylus), add the following CSS to the page:

```
.CodeMirror-cursor {
    width: 1px;
    background: rgba(255, 255, 255, 0.6) !important;
}
```

It might not do everything the cursor is supposed to do, but at least you can see it.

**Comment by SavePointSam** [comment]
Just upgraded to Apollo 2.0 and am running into this problem as well. :(

**Comment by ecwyne** [comment]
Definitely not ideal... but I figured out a way to have a visible cursor in the meantime.

1. Open the playground
2. Open the developer console and turn on [device mode](https://developers.google.com/web/tools/chrome-devtools/device-mode/#toggle)
3. Reload the page (the cursor shows up on mobile apparently)
4. Toggle device mode off and close the console.
5. Don't ever close the tab or refresh the page. 😝

**Comment by maxnachlinger** [comment]
The underline and block cursors work, might be simpler to just use those.

**Comment by jazzzz** [comment]
> The underline and block cursors work, might be simpler to just use those.

For me it did not work either.

**Comment by jameskennethrobinson** [comment]
Also having this issue with apollo server 2.0

**Comment by PieterScheffers** [comment]
The css of @ryanwiesjahn works!

To use the css every time you open the playground create a bookmark and change the url of the bookmark to this:
```
javascript:var style = document.createElement('style'); style.type='text/css'; style.appendChild(document.createTextNode('.CodeMirror-cursor { width: 1px; background: rgba(255, 255, 255, 0.6) !important; }')); document.getElementsByTagName('head')[0].appendChild(style);void(0);
```
Then when on the GraphQL playground just push this bookmark to insert the css of @ryanwiesjahn.

Tested in Google Chrome.

**Comment by massimilianomarini** [comment]
Thanks @ryanwiesjahn, with your css it is possible to use the playground! 👍

**Comment by massimilianomarini** [comment]
Yes, I can confirm what @gnemtsov said, in my Google Chrome (latest stable release) settings don't work at all. They are saved in the localStorage but are not applied, also after a hard refresh, nothing change.

**Comment by lpellegr** [comment]
I am getting the same issue as you guys when using Playground as configured by `apollo-server-express`.

The issue seems related to the default settings for the property `editor.cursorShape`. If force a value in your configuration object, then the cursor becomes visible:

```
playground: {
            settings: {
                'editor.reuseHeaders': true,
                'editor.cursorShape': 'line', // possible values: 'line', 'block', 'underline'
            }
}
```

**Comment by dandv** [comment]
Same problem when upgrading from apollo-server 2.0.0-rc5 to 2.0.0.

As a workaround, you can paste your endpoint at https://legacy.graphqlbin.com/new and use that UI. The cursor is visible there.

**Comment by acoreyj** [comment]
Not sure if the issue is apollo-server or graphql-playground but seeing the same issue. Also you can't change to the light theme

**Comment by Andrejs1979** [comment]
Same here

**Comment by iandvt** [comment]
+1 - encountered this issue when migrating from Apollo 1 -> Apollo 2

**Comment by pantharshit00** [comment]
+1

**Comment by MFQ** [comment]
+1

**Comment by 5amfung** [comment]
@lpellegr Would you mind explaining or give an example how I would force feed the setting into the configuration object?  I tried the following but it didn't work. 

```
new ApolloServer({
    schema,
    playground: {
      settings: {
        'editor.cursorShape': 'line'
      }
    },
  });
```

**Comment by rdickert** [comment]
@lpellegr strange, that code looks correct to me. It works for me with the following code on apollo-server 2.0.0:

```
  ... // set up typeDefs, resolvers, context
  const playground = {
    settings: {
      'editor.cursorShape': 'line',
    },
  };

  return new ApolloServer({ typeDefs, resolvers, context, playground });
```

Maybe #1516 above will help (if accepted)

**Comment by adostal** [comment]
@scf4 Yes. Quick hack is set settings as 'any' type.

**Comment by Alvyre** [comment]
The issue is also on koa server package :/

When you force the settings it works although

```js
playground: {
    settings: { // Force setting, workaround: https://github.com/prisma/graphql-playground/issues/790
      'editor.theme': 'dark',
      'editor.cursorShape': 'line' // possible values: 'line', 'block', 'underline'
    }
  }
```

**Comment by apollomusa** [comment]
Anyone find a fix for hapi server pkg?

**Comment by jorgeguberte** [comment]
I can replicate the issue with Opera and Chrome on Windows 10

**Comment by vincenzo** [comment]
Ha, forcing the settings works. I suspect the issue is still with Playground, though, as it seems not to be reading from the saved settings.

**Comment by thardy** [comment]
Thank you for this!  This worked for me.  I was very disappointed to try GraphQLPlayground for the first time, only to find out there is no cursor in the editor.

This solved it for me.

**Comment by DHFW** [comment]
For anyone with TypeScript errors:
```Javascript
  const playground = {
    settings: {
      'editor.cursorShape': 'line',
    } as any,
  };
```

**Comment by Huvik** [comment]
It should be finally fixed. Sorry for troubles, if somebody encounters it open it again please.

**Comment by icoderight** [comment]
I just encountered the problem. not sure what other info i can give you

**Comment by Huvik** [comment]
@icoderight What version of playground do you use?

**Comment by anthony-gallego-github** [comment]
Personally, I stopped my Dark mode extension (Dark Reader) and the cursor came back. ;p

**Comment by vxm5091** [comment]
> Personally, I stopped my Dark mode extension (Dark Reader) and the cursor came back. ;p

This one did the trick for me!
